### PR TITLE
Shell handler returns int value

### PR DESCRIPTION
**Pull request by jakub-uC**
Shell handler can now return an int value. This gives an opportunity to check if command has been executed successfully. It will allow to write tests for modules using shell commands. 

This change is needed to make new shell backward compatible with legacy shell.

### CODE REVIEW

File: subsys/shell/shell.c
Comment (nordic-krch) [inline subsys/shell/shell.c]: you could add at least log message here if shell execution fails.

File: subsys/shell/shell_cmds.c
Comment (nordic-krch) [inline subsys/shell/shell_cmds.c]: this is not that simple: precheck may also fail if help was requested. In that case return 0 is ok but if it fails because of number of arguments then it should return error.

Comment (jarz-nordic) [reply subsys/shell/shell_cmds.c]: Yes, this is good point. It will be fixed in other PR.

**Comment by nvlsianpu** [comment]
IMO changes of the looker shell should goes in dedicated commit.

File: subsys/shell/shell.c
Comment (jarz-nordic) [reply subsys/shell/shell.c]: I would like to avoid that, in my opinion logging failed execution should be implemented in command handler.

**Comment by carlescufi** [comment]
@rakons you might want to wait until this is merged for you rebase

File: include/shell/shell.h
Comment (carlescufi) [inline include/shell/shell.h]: Please use `errno.h` types instead

**Review by carlescufi — CHANGES_REQUESTED** [review]

File: include/shell/shell.h
Comment (carlescufi) [reply include/shell/shell.h]: https://github.com/zephyrproject-rtos/zephyr/blob/master/lib/libc/minimal/include/errno.h#L32

Comment (carlescufi) [reply include/shell/shell.h]: Maybe `return -ENOEXEC`

File: samples/subsys/shell/shell_module/src/main.c
Comment (Vudentz) [inline samples/subsys/shell/shell_module/src/main.c]: Shouldn't we go one step further and replace the calls to shell_cmd_precheck on the handlers with just a specific error so the shell would take care of calling shell_cmd_precheck internally?

File: subsys/shell/shell.c
Comment (nordic-krch) [reply subsys/shell/shell.c]: ok

File: include/shell/shell.h
Comment (jarz-nordic) [reply include/shell/shell.h]: done

**Review by carlescufi — APPROVED** [review]

**Comment by carlescufi** [comment]
@Vudentz @nordic-krch this should be merged soon since it will impact the transition of other subsystems to the new shell.

**Review by nordic-krch — APPROVED** [review]

**Review by Vudentz — APPROVED** [review]
LGTM

**Pull request merged**
